tools/toolchain/scripts/stage4/install_cosma.sh

```diff
@@ -144,7 +141,7 @@ case "$with_cosma" in
     ;;
 esac
 if [ "$with_cosma" != "__DONTUSE__" ]; then
-  COSMA_LIBS="-lcosma_pxgemm -lcosma -lgrid2grid IF_CUDA(-lTiled-MM|)"
+  COSMA_LIBS="-lcosma_pxgemm -lcosma"
```

Comment:
I think this line would have to be: 
```
COSMA_LIBS="-lcosma_pxgemm -lcosma -lcosta_scalapack -lcosta IF_CUDA(-lTiled-MM)"
```

Now that COSMA is a shared library, I think scalapack would also have to be built as a shared library, but I guess this is already the case?

Reply:
Not in the toolchain, no.

Reply:
@kabicm do you know whether the linker strictly adheres to the order when resolving symbols provided by both a shared and a static library or whether it moves static libraries to the top? If the library being static or dynamic changes preference we must ensure both are either static or shared I guess.

Reply:
Thanks for the hint. I've tried building scalapack as a shared library but then ELPA complained. 

I think, the best solution would be to link COSMA statically and then call `cosma::pxgemm<double>` directly in 
[cp_fm_gemm](https://github.com/cp2k/cp2k/blob/master/src/fm/cp_fm_basic_linalg.F#L362). This would keep the toolchain complexity low and also satisfy our long-standing desire to have an off-switch.

Reply:
@dev-zero this is a very good question and I am really not sure. Since i used function interpose, it requires that both scalapack and cosma are built as shared libraries.

@oschuett this is fine with me. In fact, instead of `cosma_pxgemm`, we also provide `cosma_prefixed_pxgemm` library which actually provides `cosma_pdgemm, cosma_psgemm, cosma_pcgemm` and `cosma_pzgemm` functions.